applications/adv_networking_bench/cpp/testing/adv_networking_test_harness.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import logging
+import re
+import shlex
+import subprocess
+import sys
+from subprocess import CalledProcessError, CompletedProcess
+from typing import List
+
+
+def run_bash_cmd(cmd, stream: bool = True) -> CompletedProcess:
+    """
+    Runs a bash command and captures its output.
+
+    Args:
+        cmd (str or list): The command to run.
+
+    Returns:
+        CompletedProcess: The result of the command execution.
+    """
+    logger = logging.getLogger(__name__)
+    logger.debug(f"Running command: {cmd}")
+
+    stdout_lines: List[str] = []
+    stderr_lines: List[str] = []
+
+    # Convert the bash command to a list of arguments if needed,
+    # so as not to require Shell=True
+    if isinstance(cmd, str):
+        args = shlex.split(cmd)
+    else:
+        args = cmd
+
+    # Start process
+    try:
+        p = subprocess.Popen(
+            args,
+            cwd=None,
+            universal_newlines=True,
+            bufsize=1,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+    except Exception as e:
+        logger.error(f"Failed to create process: {e}")
+        raise
+
+    # Stream & capture log
+    while p.poll() is None:
+        for line in iter(p.stderr.readline, ""):
+            if stream:
+                print(line.strip())
```

Comment:
Regarding the issue of CTest output being out-of-order, what happens if you add `flush=True` in each `print` statement, or added a second logger and aligned fully to `logging` instead of `print` functions?

Reply:
Heck ya @tbirdso. flush it is.

I didn't want to use logging and mixup log formats from spdlog vs python, not explicitly call `logger.err` and `logger.info` since for some reason, all of holoscan's output is to stderr and not stdout.

Reply:
See https://github.com/nvidia-holoscan/holohub/compare/e12d8e5fd48c24bb2d4b6e8fd68481ae261f8b2e..1999ee7f4087ca448b971473c118a6c5132f073c#diff-7681f135bbdea083404294061cb305b91e8ba0c18667bdcaca7b4c2391ff4807R66